Objects/bytesobject.c

```diff
@@ -2968,11 +2922,12 @@ PyBytes_FromObject(PyObject *x)
     if (PyObject_CheckBuffer(x))
         return _PyBytes_FromBuffer(x);
 
-    if (PyList_CheckExact(x))
-        return _PyBytes_FromList(x);
-
-    if (PyTuple_CheckExact(x))
-        return _PyBytes_FromTuple(x);
+    if (PyList_CheckExact(x) || PyTuple_CheckExact(x)) {
+        PyObject *bytes = _PyBytes_FromSequence(x);
+        if (bytes != Py_None) {
```

Comment:
Just throw a comment saying that Py_None is used for falling back to the slow path.